src/API/Google/AdsCampaign.php

```diff
@@ -388,6 +409,42 @@ protected function convert_campaign( GoogleAdsRow $row ): array {
 		return $data;
 	}
 
+	/**
+	 * Combine converted campaigns data with campaign criterion results data
+	 *
+	 * @param array             $campaigns                  Campaigns data returned from a query request and converted by convert_campaign function.
+	 * @param PagedListResponse $campaign_criterion_results Campaigns criterion results returned from a query request.
+	 *
+	 * @return array
+	 */
+	protected function combine_campaigns_and_campaign_criterion_results( array $campaigns, PagedListResponse $campaign_criterion_results ): array {
+		$campaign_ids = array_column( $campaigns, 'id' );
+
+		/** @var GoogleAdsRow $row */
+		foreach ( $campaign_criterion_results->iterateAllElements() as $row ) {
+			$campaign           = $row->getCampaign();
+			$campaign_id        = $campaign->getId();
+			$campaign_criterion = $row->getCampaignCriterion();
+			$location           = $campaign_criterion->getLocation();
+
+			// TODO: Convert the geo_traget_constant to the ISO 3166-1 alpha-2 country code
+			// after https://github.com/woocommerce/google-listings-and-ads/issues/1229 is done.
+			$geo_target_constant = $location ? $location->getGeoTargetConstant() : null;
+
+			$campaign_key = array_search( $campaign_id, $campaign_ids, true );
+
+			if ( ! array_key_exists( 'targeted_locations', $campaigns[ $campaign_key ] ) ) {
+				$campaigns[ $campaign_key ]['targeted_locations'] = [];
+			}
+
+			if ( ! empty( $geo_target_constant ) ) {
+				$campaigns[ $campaign_key ]['targeted_locations'][] = $geo_target_constant;
+			}
```

Comment:
❓ / 🚧  

I think we should check here if the `$campaign_key` was found because `array_search` will return false in the case is not found but the rest of the code in lines 436, 440 still runs so it will add all the targeted locations in `$campaigns[false] `

Since in PHP `$campaigns[false]`  evaluates to `$campaigns[0] ` it will create potential undesired behavioiur in this function.

Reply:
Ah, that's indeed a potential bug, updated in 551c74b, thanks.